### ISSUE TITLE
Migrate designs to website

### ISSUE THREAD

**Issue by jaspercooper** [DeclareDesign/DesignLibrary#70]
- [ ] no more designs in the library package
- [ ] create `download_design()` function
- [ ] create test for website designs

**Comment by jaspercooper** [comment]
One idea is to create a DeclareDesign dataverse and host designs there

**Comment by graemeblair** [comment]
I agree versioning would be good, suggest Github is probably better/easier solution?

**Comment by nfultz** [comment]
GitHub is really terrible for binary files.

**Comment by graemeblair** [comment]
fair enough. I think we want something with an R api that is easy to use, so we can access it within DesignLibrary. 

dataverse has one, but I've had to fight with dataverse every time I've used it.

**Comment by jaspercooper** [comment]
Yeah I know what you mean about dataverse API. I was hoping this had gotten better.

One advantage I see in setting up a DD dataverse is that it contains a lot of the infrastructure we already want, and would prevent people uploading any old binary file to our website. If we end up going with something like AWS to host the .Rdata files we'll need some method of reviewing and approving uploads. 

Dataverse has both versioning and review: http://guides.dataverse.org/en/latest/user/dataset-management.html

Definitely need to overcome the API problem though I agree. Accessing designs from inside the package needs to be seamless.

**Comment by nfultz** [comment]
https://github.com/IQSS/dataverse-client-r/issues/21 - might be churning.

**Comment by graemeblair** [comment]
OSF also has an API that is more likely to be maintained, but no R package. Problem with dataverse is not just the R package.

**Comment by pdurbin** [comment]
@graemeblair I'm so sorry to hear you've had to fight with the Dataverse API. If you would be willing to let me know here or open issues over at https://github.com/IQSS/dataverse/issues or pop in http://chat.dataverse.org I'd love to get some feedback from you on them. We are continually making improvements, such as easier API for making small metadata edits: https://github.com/IQSS/dataverse/issues/3599 . I also recently added headers to the "native" API documentation so you can more easily tell what the API can even do: http://guides.dataverse.org/en/4.9/api/native-api.html

Yes, there in churn in dataverse-client-r in the sense that the original author (whom we love!) is switching jobs and looking for maintainers for a number of packages that he has been maintaining for years: https://twitter.com/thosjleeper/status/994135444848889856

**Comment by graemeblair** [comment]
thanks @pdurbin I haven't tried in awhile so will try again!
